### PR TITLE
Update jupyter-book version

### PR DESCRIPTION
**Pull request by tlambert03**
This PR just bumps the version of jupyter-book that we expect when building docs (from 0.10.x to 0.12.1) ... there are a few features (like sphinx-tabs) that I'd like to use that are only compatible with newer versions.

Requires only a slight TOC reformat

### CODE REVIEW

**Review by nclack — APPROVED** [review]
lgtm

**Comment by tlambert03** [comment]
thanks @nclack ... @codemonkey800, @kne42, @melissawm, @GenevieveBuckley... do any of you have any objections, or know of any issues that might arise from bumping this version?

File: Makefile [hunk 1]
```diff
@@ -6,9 +6,7 @@ docs:
 	pip install -qr docs/requirements.txt
 	python docs/_scripts/update_preference_docs.py
 	python docs/_scripts/update_event_docs.py
-	NAPARI_APPLICATION_IPY_INTERACTIVE=0
-	jb build docs
-	unset NAPARI_APPLICATION_IPY_INTERACTIVE
+	NAPARI_APPLICATION_IPY_INTERACTIVE=0 jb build docs
```
Comment (jni) [inline Makefile:9]: :+1:

**Review by jni — APPROVED** [review]
@tlambert03 just to double check, this builds fine locally? I can't remember the differences between article and book but my impression was that "book" was a closer fit with the default format before?

**Comment by tlambert03** [comment]
> @tlambert03 just to double check, this builds fine locally?

yep, been using it while testing the npe2 docs

**Comment by tlambert03** [comment]
>  but my impression was that "book" was a closer fit with the default format before?

but lemme check book too :)  hang on a min

**Comment by tlambert03** [comment]
ok, so, it looks like jupyter-book suggested `article` when auto-converting the toc based on the headers we were using. I had to change the top level  `sections` to `chapters` to use the book format, but after doing so it works fine.  pushed.

**Comment by jni** [comment]
> it looks like jupyter-book suggested article when auto-converting the toc based on the headers we were using. I had to change the top level sections to chapters to use the book format, but after doing so it works fine

Yeah I had a bit of a discussion on this with the jbook folks [here](https://github.com/executablebooks/jupyter-book/issues/1355), but it was never resolved...

Anyway, my approval stands either way. :stuck_out_tongue_winking_eye:

**Review by Carreau — APPROVED** [review]

**Comment by Carreau** [comment]
Let's try.

**Pull request merged**